tools/add_triage_signature.py

```diff
@@ -325,6 +333,78 @@ def _update_ticket(self, url, issue_key, should_update=False):
         self._update_description(issue_key, description)
 
 
+class FailureDetails(Signature):
+    def __init__(self, jira_client):
+        super().__init__(jira_client, comment_identifying_string="")
+
+    def is_olm_operator(self, operator_name):
+        return operator_name.lower() in ["cnv", "lso", "ocs"]
+
+    def _update_ticket(self, url, issue_key, should_update=False):
+
+        #import ipdb; ipdb.set_trace()
+        def extract_cluster_md_from_existing_labels():
+            label_prefix_to_property = {
+                "AI_CLUSTER_": "id",
+                "AI_DOMAIN_": "email_domain",
+                "AI_USER_": "user_name",
+            }
+            r = re.compile(r'(AI_[^_]*_)(.*)')
+
+            i = self._jclient.issue(issue_key)
+            cluster_md = {}
+            for l in i.fields.labels:
+                m = r.match(l)
+                if m is None or len(m.groups()) != 2:
+                    continue
+                prefix, value = m.groups()
+                if prefix not in label_prefix_to_property:
+                    continue
+                cluster_md[label_prefix_to_property[prefix]] = value
+            return cluster_md
```

Comment:
extract as method

Reply:
it is a method :)

Reply:
extract_cluster_md_from_existing_labels is an inner function
I think it will be clearer to get it out of the outer method

Reply:
I don't agree
It is specific for this "signature" and also uses internals state of this signature